sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -186,8 +203,12 @@ public DefaultAzureCredential build() {
     }
 
     private ArrayList<TokenCredential> getCredentialsChain() {
-        ArrayList<TokenCredential> output = new ArrayList<TokenCredential>(7);
+        WorkloadIdentityCredential workloadIdentityCredential = (CoreUtils.isNullOrEmpty(managedIdentityClientId) && CoreUtils.isNullOrEmpty(workloadIdentityClientId)) ?  null : getWorkloadIdentityCredentialIfAvailable();
```

Comment:
Looks like we should unconditionally call the helper. When neither ClientID option is set AZURE_CLIENT_ID could still have a value; if it does, we should use that value.

Reply:
Default value of managedIdentityClientId is value in AZURE_CLIENT_ID env var.
User can override it by setting on the builder.
so, the env var value check is captured automatically here.

Reply:
I see. It's maybe a good idea to remove this conditional anyway, unless the method call is expensive, because the helper checks the same conditions.

Reply:
removed this conditional check.